pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -118,6 +123,18 @@ func validateBundlesRefCluster(new, old *Cluster) field.ErrorList {
 	return allErrs
 }
 
+// ValidateControlPlaneIPUniqueness checks whether or not the control plane endpoint defined
+// in the cluster spec is available.
+func ValidateControlPlaneIPUniqueness(cluster *Cluster, netClient networkutils.NetClient) error {
```

Comment:
Does this need to be exported?

Reply:
It looks like it does which makes me think this is the wrong package to contain this function. It could reside in a `validate` package (or whatever already exists with this little funcs if we have one).

Reply:
> I think we might run into an import cycle since the first parameter requires the v1alpha1 package. Because if we move it elsewhere, we would still need to call that method here in the v1alpha1 package since we want to validate things at the API level for the new controller.

Good catch, that's right. 

My gripe is with the snow package calling a validate function in the API package - that just seems odd to me but I suppose framed the right way it could make sense. Thanks for humoring me.